sqlflow.proto

```diff
@@ -1,49 +1,51 @@
 syntax = "proto3";
 
+package main;
+
 service SQLFlow {
-  rpc Run(RunRequest) returns (RunReply) {}
+  // Run runs a SQL statement and returns a stream of RunReply.  Some
+  // SQL statements like `USE database` returns only a success status.
+  // In such cases, the return stream contains only one RunReply with
+  // RunReply.error set.  Some other SQL statements like `SELECT ...`
+  // returns a table in addition to the status, and the table might be
+  // big.  In such cases, Run returns a stream of RunReply -- the
+  // first one has RunReply.schema set, and all RunReplies have
+  // RunReply.error and RunReply.columns set -- each RunReply doesn't
+  // exceed the maximum message size
+  // (https://stackoverflow.com/a/34186672/724872).
+  rpc Run(RunRequest) returns (stream RunReply) {}
 }
 
 message RunRequest {
   string sql = 1;		// The SQL statement to be executed.
 }
 
 message RunReply {
-  string error = 1;	 // An empty string indicates no error.
-  Table table  = 2;	 // `show databases` and `select` return data.
+  string error = 1;
+  map<string, FieldType> schema = 2; // only set if RunRequest.sql returns data.
+  map<string, Column> columns = 3; // only set if RunRequest.sql returns data.
```

Comment:
It may use one of has {
    repeated string messages = 1;
   map<string, Column> columns = 2;
}

to support both data and log message, so client may show progress to users for long running job.

Reply:
columns may put into another message type so it can support pagination later without breaking compatibility. same as messages.

Reply:
这地方没明白。能写一下例子吗？

Reply:
Done.